source/server/listener_impl.cc

```diff
@@ -123,15 +123,85 @@ Network::SocketSharedPtr ListenSocketFactoryImpl::getListenSocket() {
   return createListenSocketAndApplyOptions();
 }
 
+ListenerFactoryContextBaseImpl::ListenerFactoryContextBaseImpl(
+    Envoy::Server::Instance& server, ProtobufMessage::ValidationVisitor& validation_visitor,
+    const envoy::config::listener::v3::Listener& config, DrainManagerPtr drain_manager)
+    : server_(server), metadata_(config.metadata()), direction_(config.traffic_direction()),
+      global_scope_(server.stats().createScope("")),
+      // Not ideal
```

Comment:
? Comment not actionable.

Reply:
I remember I resolve addresses multiple times... NBD. Deleted the comment and we will rediscover if it matters